src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -467,7 +467,7 @@ namespace cryptonote
     m_blockchain_storage.set_user_options(blocks_threads,
         blocks_per_sync, sync_mode, fast_sync);
 
-    r = m_blockchain_storage.init(db, m_testnet, test_options);
+    r = m_blockchain_storage.init(db.release(), m_testnet, test_options);
```

Comment:
I don't think this is entirely sufficient. Inside Blockchain::init() there is an error path (Line 321) that doesn't free up the memory. My understanding is after Blockchain::init()::327 there is a guarantee that the deinit() function will be called and clean it up. 

Am I following that correctly?

Reply:
You seem to be correct, thanks.